### PR TITLE
feat(regExp): Search bar name regex and error component

### PR DESCRIPTION
**Pull request by atticusofsparta**
image
Screenshot from 2022-11-10 11-01-22

### CODE REVIEW

File: src/components/inputs/Search/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -17,30 +19,44 @@ function SearchBar(props: SearchBarProps) {
...
+        e.target.value.length > 1)
+    ) {
+      setSearchBarText(e.target.value);
+      setIsValid(true);
+    } else {
+      setIsValid(false);
+    }
     if (e.target.value === '') {
       reset();
     }
```
Comment (dtfiedler) [inline src/components/inputs/Search/SearchBar/SearchBar.tsx:42]: put this first and return after, then you can remove the <1 parameters I believe

File: src/components/inputs/Search/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -49,15 +65,17 @@ function SearchBar(props: SearchBarProps) {
...
+        isAvailable,
       })}
       <div
         className="searchBar"
         style={
-          !searchSubmitted || isDefault
-            ? { borderColor: '' }
-            : isValid
-            ? { borderColor: 'var(--success-green)' }
+          isValid
```
Comment (dtfiedler) [inline src/components/inputs/Search/SearchBar/SearchBar.tsx:73]:
```suggestion
          isAvailable
```

File: src/components/inputs/Search/SearchBar/SearchBar.tsx [hunk 3]
```diff
@@ -84,6 +104,19 @@ function SearchBar(props: SearchBarProps) {
       </div>
       {React.cloneElement(footerElement, {
         ...props,
+        defaultText: (
+          <>
+            {!isValid ? (
```
Comment (dtfiedler) [inline src/components/inputs/Search/SearchBar/SearchBar.tsx:109]: let's pass this to the `SearchBarFooter` element and put this div in there

File: src/components/inputs/Search/SearchBar/SearchBar.tsx (hunk 1 above)
Comment (atticusofsparta) [reply src/components/inputs/Search/SearchBar/SearchBar.tsx:42]: no still need that for if a user wants to completely delete the string. The second <1 is to allow dashes to be input on the end, which fails the regex, but is caught in the onSubmit function instead.

File: src/components/inputs/Search/SearchBar/SearchBar.tsx (hunk 3 above)
Comment (atticusofsparta) [reply src/components/inputs/Search/SearchBar/SearchBar.tsx:109]: added

File: src/components/modals/ConnectWalletModal/ConnectWalletModal.tsx [hunk 4]
```diff
@@ -49,14 +49,15 @@ function ConnectWalletModal({
           <img src={ArweaveAppIcon} alt="" width="47px" height="47px" />
           Connect using Arweave.app
         </button>
-        <text className="textFaded" style={{ marginTop: '1em' }}>
-          Don't have a wallet? get one{' '}
+        <text className="textWhite" style={{ marginTop: '1em' }}>
```
[the remark this answers is not in the document]
Comment (atticusofsparta) [reply src/components/modals/ConnectWalletModal/ConnectWalletModal.tsx:52]: broke down some classes here, will continue to follow this pattern as I build more and come across them.

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**